crates/fluvio-spu/src/mirroring/remote/controller.rs

```diff
@@ -0,0 +1,523 @@
+                .await?;
+            Ok(())
+        } else {
+            Ok(())
+        }
+    }
+
+    /// look up home cluster from local store
+    /// this may retur None if remote cluster is send by SC by time controller is started
+    fn find_home_cluster(&self) -> Option<Home> {
+        let read = self.mirror_store.read();
+        let mirror = read.get(&self.remote_config.home_cluster).cloned();
+
+        match mirror {
+            Some(mirror) => match mirror.spec.mirror_type {
+                MirrorType::Home(home) => Some(home),
+                _ => None,
+            },
+            None => None,
+        }
+    }
+
+    /// compute records necessary to fill in gap for mirror home
+    async fn generate_home_sync(
+        &self,
+        home_leo: Offset,
+    ) -> Result<Option<FilePartitionSyncRequest>> {
+        // leader off should be always greater than remote leo
+        let leader_offset = self.leader.as_offset();
+
+        // if remote mirror is all caught up, there is no need to send out update
+        if leader_offset.leo == home_leo {
+            debug!("home has caught up, just chilling out");
+            return Ok(None);
+        }
+
+        let mut partition_response = FilePartitionSyncRequest {
+            leo: leader_offset.leo,
+            hw: leader_offset.hw,
+            ..Default::default()
+        };
+
+        if leader_offset.leo > home_leo {
+            match self
+                .leader
+                .read_records(home_leo, self.max_bytes, self.isolation)
+                .await
+            {
+                Ok(slice) => {
+                    debug!(
+                        hw = slice.end.hw,
+                        leo = slice.end.leo,
+                        replica = %self.leader.id(),
+                        "read records"
+                    );
+                    if let Some(file_slice) = slice.file_slice {
+                        partition_response.records = file_slice.into();
+                    }
+                    Ok(Some(partition_response))
+                }
+                Err(err) => {
+                    error!(%err, "error reading records");
+                    Err(anyhow!("error reading records: {}", err))
+                }
+            }
+        } else {
+            //
+            debug!(
+                hw = leader_offset.hw,
+                leo = leader_offset.leo,
+                home_leo,
+                "oh no mirror home has more records"
+            );
+            Err(anyhow!(
+                "leader has more records than home, this should not happen"
+            ))
+        }
+    }
+
+    /// create socket to home, this will always succeed
+    #[instrument]
+    async fn create_socket_to_home(
+        &self,
+        backoff: &mut ExponentialBackoff,
+        _home: &Home,
+    ) -> (FluvioSocket, bool) {
+        //TODO: remove the comments bellow when we have tls support
```

Comment:
We should not use comments as code storage.

Reply:
Agree! Fixed.